WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -439,163 +441,246 @@ def register
     end
   end
 
+  # Respond to asynchronous payment updates from Stripe.
+  # Code skeleton according to https://stripe.com/docs/webhooks/quickstart
+  def stripe_webhook
+    payload = request.body.read
+
+    begin
+      event = Stripe::Event.construct_from(
+        JSON.parse(payload, symbolize_names: true),
+      )
+    rescue JSON::ParserError => e
+      # Invalid payload
+      logger.warn "Stripe webhook error while parsing basic request. #{e.message}"
+      return head :bad_request
+    end
+    # Check if webhook signing is configured.
+    if EnvVars.STRIPE_WEBHOOK_SECRET.present?
+      # Retrieve the event by verifying the signature using the raw body and secret.
+      signature = request.env['HTTP_STRIPE_SIGNATURE']
+      begin
+        event = Stripe::Webhook.construct_event(
```

Comment:
nice!

Reply:
Thanks although this is literally me copying Stripe documentation suggestions